src/mezz/base-series.r

```diff
@@ -243,23 +285,11 @@ trim: function [
                 fail "Invalid refinements for TRIM of BINARY!"
             ]
 
-            rule: case [
-                with and (binary? str) [
-                    ;
-                    ; !!! MAKE BITSET! of a BINARY! doesn't treat it as a set
-                    ; of bytes, rather as the raw data underlying a bitset.
-                    ; Work around it by using MAKE BITSET! on an array of
-                    ; integer! values.
-                    ;
-                    ; !!! Can't use COLLECT because that's in "Mezzanine" and
-                    ; this is in "Base".  Review these locations.
-                    ;
-                    array: make block! length of str
-                    for-each b str [append array b]
-                    make bitset! array
-                ]
-                with [make bitset! str]
-            ] else [#{00}]
+            rule: either with [
+                either bitset? str [str] [charset str]
+            ][
+                #{00}
+            ]
```

Comment:
I'm curious how comfortable people can get with ?? and !!, and mixing them with other branching constructs:

    rule: if with [either bitset? str [str] [charset str]] !! #{00}

If one warms up to them, I think they are pretty nice.